rust/operator-binary/src/config/jvm.rs

```diff
@@ -0,0 +1,205 @@
+use snafu::{ResultExt, Snafu};
+use stackable_hdfs_crd::{constants::JVM_SECURITY_PROPERTIES_FILE, HdfsCluster, HdfsRole};
+use stackable_operator::{
+    k8s_openapi::api::core::v1::ResourceRequirements,
+    memory::{BinaryMultiple, MemoryQuantity},
+    role_utils::JvmArgumentOverrides,
+};
+
+const JVM_HEAP_FACTOR: f32 = 0.8;
+
+#[derive(Snafu, Debug)]
+pub enum Error {
+    #[snafu(display("invalid java heap config for {role:?}"))]
+    InvalidJavaHeapConfig {
+        source: stackable_operator::memory::Error,
+        role: String,
+    },
+
+    #[snafu(display("failed to merge jvm argument overrides"))]
+    MergeJvmArgumentOverrides { source: stackable_hdfs_crd::Error },
+}
+
+// All init or sidecar containers must have access to the following settings.
+// As the Prometheus metric emitter is not part of this config it's safe to use for hdfs cli tools as well.
+// This will not only enable the init containers to work, but also the user to run e.g.
+// `bin/hdfs dfs -ls /` without getting `Caused by: java.lang.IllegalArgumentException: KrbException: Cannot locate default realm`
+// because the `-Djava.security.krb5.conf` setting is missing
+pub fn construct_global_jvm_args(kerberos_enabled: bool) -> String {
+    let mut jvm_args = Vec::new();
+
+    if kerberos_enabled {
+        jvm_args.push("-Djava.security.krb5.conf=/stackable/kerberos/krb5.conf".to_owned());
```

Comment:
Not your doing, but can we make `/stackable/kerberos/` a constant? Its used in container.rs as well a couple of times for env vars etc.

Reply:
It was actually my doing :D I personally intentionally decided against templating it everywhere to avoid complexity. However, I will push a commit to template it

Reply:
I think i rather template and have it consistent, but leave it up to you :)